infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -790,7 +797,7 @@ public void handle(PodEvent event) {
         for (Entry<String, KubernetesMachineImpl> entry :
             machines.getMachines(getContext().getIdentity()).entrySet()) {
           final KubernetesMachineImpl machine = entry.getValue();
-          if (machine.getPodName().equals(podName)) {
+          if (podName.startsWith(machine.getPodName())) {
```

Comment:
Ah, sorry I misunderstood. I'm not sure `equals` is sufficient here -- in `KubernetesDeployments`, `podName` is set to the involved object's `metadata.name`. For workspace pods, this will be `machine.getPodName()` plus a suffix added by the replica set.

Reply:
> podName is set to the involved object's metadata.name

It's true

> For workspace pods, this will be `machine.getPodName()` plus a suffix added by the replica set.

I guess `machine.getPodName()` will return the value with suffix but not configured one. I think so because be create a machine with podName that is fetched from a deployed pod.
Here is a place where we fetch deployed pod 
https://github.com/eclipse/che/blob/07263f1e30089689d71b057f747a44a29283e3c4/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java#L513
And here we use its name
https://github.com/eclipse/che/blob/07263f1e30089689d71b057f747a44a29283e3c4/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java#L524

Correct me If I am wrong and deployed pod has a name without a suffix added by the replica set.

Reply:
Ah I see what you mean now -- you're right. I've reverted to using `equals` here.